test/cli-test.js

```diff
@@ -0,0 +1,70 @@
+/* global describe, it */
+var assert = require('assert').strict
+const { exec } = require('child_process')
+
+function cli (command) {
+  return new Promise((resolve, reject) => {
+    exec(command, (error, stdout, stderr) => {
+      if (error) {
+        reject(error)
+      } else {
+        resolve({
+          exitCode: 0,
+          out: JSON.parse(stdout)
+        })
+      }
+    })
+  })
+}
+
+var tests = [
+  /* {
+  c: `node ${__dirname}/../lib/cli.js "kind" ${__dirname}/fixture/activities.json`,
+  e: {
+    exitCode: 0,
+    out: {
+      kind: 'plus#activity'
+    }
+  }
+  }, */
+  {
+    c: `cat ${__dirname}/fixture/activities.json | node ${__dirname}/../bin/json-mask.js "kind"`,
+    e: {
+      exitCode: 0,
+      out: {
+        kind: 'plus#activity'
+      }
+    }
+  }, {
+    c: `cat ${__dirname}/fixture/activities.json | node ${__dirname}/../bin/json-mask.js "object(objectType)"`,
+    e: {
+      exitCode: 0,
+      out: {
+        object: { objectType: 'note' }
+      }
+    }
+  }, {
+    c: `cat ${__dirname}/fixture/activities.json | node ${__dirname}/../bin/json-mask.js "url,object(content,attachments/url)"`,
+    e: {
+      exitCode: 0,
+      out: {
+        url: 'https://plus.google.com/102817283354809142195/posts/F97fqZwJESL',
+        object: {
+          content: 'Congratulations! You have successfully fetched an explicit public activity. The attached video is your reward. :)',
+          attachments: [{ url: 'http://www.youtube.com/watch?v=dQw4w9WgXcQ' }]
+        }
+      }
+    }
+  }]
+
+describe('cli', function () {
+  var result, i
+  for (i = 0; i < tests.length; i++) {
+    (function (test) {
+      it('should mask with ' + test.c + ' in test #' + i, async function () {
+        result = await cli(test.c)
+        assert.deepStrictEqual(result, test.e)
+      })
+    }(tests[i]))
+  }
+})
```

Comment:
Overall the coverage is still pretty low on `bin/json-mask`, especially on branches. If you can, try covering as much as possible.

```
---------------|----------|----------|----------|----------|-------------------|
File           |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------|----------|----------|----------|----------|-------------------|
All files      |    90.07 |    79.52 |    95.83 |    92.42 |                   |
 bin           |    64.29 |     12.5 |    85.71 |    61.54 |                   |
  json-mask.js |    64.29 |     12.5 |    85.71 |    61.54 |... 14,15,31,32,34 |
---------------|----------|----------|----------|----------|-------------------|
```

Reply:
A sure, didn't have this on my radar. node-tap runs coverage for me automatically on each test. Will bring this up to 100%.